File: src/main/java/com/vaadin/demo/component/slider/SliderBasicFeatures.java
package com.vaadin.demo.component.slider;

import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.slider.DecimalSlider;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("slider-basic-features")
public class SliderBasicFeatures extends Div {

    public SliderBasicFeatures() {
        // tag::snippet[]
        DecimalSlider slider = new DecimalSlider("Label");
        slider.setHelperText("Helper text");
        add(slider);
        // end::snippet[]
    }

    public static class Exporter extends DemoExporter<SliderBasicFeatures> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/slider/slider-basic-features.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/slider';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

export @customElement('slider-basic-features')
class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-slider label="Label" helper-text="Helper text"></vaadin-slider>
      <!-- end::snippet[] -->
    `;
  }
}
